player.js

```diff
@@ -0,0 +1,96 @@
+const LainPlayer = (() => {
+    const playerHTML = "<a id='play-toggle' onclick='LainPlayer.play()'><i class='fa fa-play' aria-hidden='true'></i></a><i id=volume-button class='fa fa-volume-up' aria-hidden='true' onclick='LainPlayer.toggleMute()'></i><input type='range' value=0.51 min=0 max=1  step=0.01 oninput='LainPlayer.changeVolume(this.value)'/>"
+    const audioContext = new window.AudioContext();
+    const audioTag = document.getElementById("audio");
+    let muted = false;
+    let volume = 0.51;
+    audioTag.volume =  volume;
+
+    function replacePlayer() {
+        const oldPlayer = audioTag;
+        const newPlayer = document.createElement("div")
+        newPlayer.setAttribute("id", "lainplayer");
+        newPlayer.innerHTML = playerHTML;
+        // hide old player and add the new one
+        oldPlayer.setAttribute("style", "display:none;");
+        oldPlayer.after(newPlayer);
+
+    }
+
+    function cangeSource(source) {
+        console.log("fug");
+        const paused = audioTag.paused;
+        console.log(paused)
+        // Use either the ogg or mp3 stream, depending on what the current one is.
+        if(audioTag.currentSrc.endsWith("ogg")) {
+            audioTag.src = "/radio/" + source + ".ogg";
+        } else {
+            audioTag.src = "/radio/" + source + ".mp3";
+        }
+
+        // Load the new audio stream.
+        audioTag.load();
+
+        // Start playing, if it was before.
+        if(!paused) {
+            audioTag.play();
+        }
+    }
+
+    function updatePlayButton() {
```

Comment:
This function should just modify the class list of the button, rather than alter its HTML.